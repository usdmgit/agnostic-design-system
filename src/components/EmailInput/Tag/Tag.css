.tag-container {
  align-items: var(--tag-align-items, normal);
  background-color: var(--tag-background-color, var(--structure-color-100));
  border-radius: var(--tag-border-radius, 4px);
  color: var(--tag-color, var(--structure-color-600));
  display: var(--tag-display, block);
  font-family: var(--tag-font-family);
  font-size: var(--tag-font-size, 16px);
  font-weight: var(--tag-font-weight, 400);
  justify-content: var(--tag-justify-content, normal);
  margin: var(--tag-margin, auto);
  padding: var(--tag-padding, 4px);
  text-transform: var(--tag-text-transform, normal);
  width: var(--tag-width, fit-content);
}
