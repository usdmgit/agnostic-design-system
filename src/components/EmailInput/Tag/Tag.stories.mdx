# Tag Component

## Type: Simple

## Tag Props

|        Prop        |      Type       | Default | Mandatory |
| :----------------: | :-------------: | :-----: | :-------: |
|      children      | React.ReactNode |    -    |   true    |
| variablesClassName |     string      |    -    |   false   |

## Getting Started

```javascript
import React from 'react';
import { Tag } from '@codelittinc/agnostic-design-system';

const MyComponent = () => {
  return (
    <div>
      <Tag>Tag Name</Tag>
    </div>
  );
};

export default MyComponent;
```

## CSS Variables Available

| Property  |    Attribute     | State |
| :-------: | :--------------: | :---: |
|           |    align-items   |   -   |
|           | background-color |   -   |
|           |  border-radius   |   -   |
|           |      color       |   -   |
|           |     display      |   -   |
|           |   font-family    |   -   |
|           |    font-size     |   -   |
|           |   font-weight    |   -   |
|           | justify-content  |   -   |
|           |      margin      |   -   |
|           |  text-transform  |   -   |
|           |      width       |   -   |
