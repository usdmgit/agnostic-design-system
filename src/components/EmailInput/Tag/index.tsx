import React from 'react';
import styles from './Tag.css';
import classNames from 'classnames';

type TagProps = {
  children: React.ReactNode;
  variablesClassName?: string;
};

const Tag: React.FC<TagProps> = (props: TagProps) => {
  const { children, variablesClassName } = props;

  return <div className={classNames(styles['tag-container'], variablesClassName)}>{children}</div>;
};

export default Tag;
