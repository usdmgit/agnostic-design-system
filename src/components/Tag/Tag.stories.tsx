import React from 'react';
import Tag from '@/components/Tag';
import { Meta } from '@storybook/react';
import mdx from './Tag.stories.mdx';

export default {
  title: 'Components/Tag',
  component: Tag,
  parameters: {
    docs: {
      page: mdx
    }
  }
} as Meta;

const Template = args => {
  return <Tag {...args} />;
};

export const Default = Template.bind({});
Default.args = {
  children: 'Testing Tag'
};

export const WithHTMLChildren = Template.bind({});
WithHTMLChildren.args = {
  children: <div>Tag with a HTML children</div>
};
