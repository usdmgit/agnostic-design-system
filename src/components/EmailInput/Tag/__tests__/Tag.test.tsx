import React from 'react';
import Tag from '@/components/Tag';
import { mount } from 'enzyme';

describe('Tag', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Tag>Testing Tag</Tag>);
    expect(wrapper).toMatchSnapshot();
  });
});
