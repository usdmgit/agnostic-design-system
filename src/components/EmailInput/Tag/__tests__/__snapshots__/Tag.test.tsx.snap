// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly 1`] = `
<Tag>
  <div
    className="tag-container"
  >
    Testing Tag
  </div>
</Tag>
`;
